Update static analyzer bot scripts

briannafan opened the pull request:
#### abd7e7b3bbfd3fafab8efb1adb4501438cc1f182
<pre>
Update static analyzer bot scripts
<a href="https://bugs.webkit.org/show_bug.cgi?id=273857">https://bugs.webkit.org/show_bug.cgi?id=273857</a>
<a href="https://rdar.apple.com/127705479">rdar://127705479</a>

Reviewed by Ryosuke Niwa and David Kilzer.

Changes logging to be concise and readable.
Adds NoUncountedMemberChecker and RefCntblBaseVirtualDtor to the checker lists.

* Tools/Scripts/compare-static-analysis-results: Renamed from Tools/Scripts/compare-static-analysis-results.py.
(parser):
(find_diff):
(compare_project_results):
(create_filtered_results_dir):
(main):
* Tools/Scripts/generate-dirty-files: Renamed from Tools/Scripts/generate-dirty-files.py.
(parser):
(parse_results_file):
(find_project_results):
(find_all_results):
(main):

Canonical link: <a href="https://commits.webkit.org/278523@main">https://commits.webkit.org/278523@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f7f50069be301bbc895613e4bba7feca08e92705

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/50860 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30158 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3178 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/54119 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/1551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/53163 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36433 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1203 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/54119 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/52959 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/27799 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/43814 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/54119 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/25143 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/1070 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-skia~~](https://ews-build.webkit.org/#/builders/52/builds/9301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/1132 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/55713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/25962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/1016 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/55713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/27219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/43889 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/55713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/28087 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7363 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/26951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Tools/Scripts/compare-static-analysis-results:
```diff
@@ -94,19 +100,18 @@ def compare_project_results(args, archive_path, new_path, project):
+        print(f'    Issues fixed: {len(fixed_issues)}')
+        print(f'    Files fixed: {len(fixed_files)}')
+        print(f'    New issues: {len(new_issues)}')
+        print(f'    New dirty files: {len(new_files)}')
```
I don't think we wanna use this terminology in developer facing UI. Probably says "files with new issues" or something.

ddkilzer commented on Tools/Scripts/compare-static-analysis-results:
```diff
@@ -26,7 +26,7 @@
 import sys
 
-CHECKERS = ['UncountedCallArgsChecker', 'UncountedLocalVarsChecker']
+CHECKERS = ['UncountedCallArgsChecker', 'UncountedLocalVarsChecker', 'NoUncountedMemberChecker', 'RefCntblBaseVirtualDtor']
```
Nit:  Keep this in alphabetical order.

ddkilzer commented on Tools/Scripts/generate-dirty-files:
```diff
@@ -29,7 +29,9 @@
-    'Uncounted raw pointer or reference not provably backed by ref-counted variable': 'UncountedLocalVarsChecker'
+    'Uncounted raw pointer or reference not provably backed by ref-counted variable': 'UncountedLocalVarsChecker',
+    'Member variable is a raw-pointer/reference to reference-countable type': 'NoUncountedMemberChecker',
+    "Reference-countable base class doesn't have virtual destructor": 'RefCntblBaseVirtualDtor'
```
Nit:  Alphabetize by the checker name (even though it's the second item).

ddkilzer reviewed (approved):
r=me.  You do not have to fix the nits—will just make it easier for future humans to make changes if the items are sorted alphabetically.

briannafan committed:
Update static analyzer bot scripts
https://bugs.webkit.org/show_bug.cgi?id=273857
rdar://127705479

Reviewed by Ryosuke Niwa and David Kilzer.

Changes logging to be concise and readable.
Adds NoUncountedMemberChecker and RefCntblBaseVirtualDtor to the checker lists.

* Tools/Scripts/compare-static-analysis-results: Renamed from Tools/Scripts/compare-static-analysis-results.py.
(parser):
(find_diff):
(compare_project_results):
(create_filtered_results_dir):
(main):
* Tools/Scripts/generate-dirty-files: Renamed from Tools/Scripts/generate-dirty-files.py.
(parser):
(parse_results_file):
(find_project_results):
(find_all_results):
(main):

Canonical link: https://commits.webkit.org/278523@main
